compiler/rustc_const_eval/src/interpret/terminator.rs

```diff
@@ -115,7 +115,72 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 }
             }
 
-            TailCall { func: _, args: _, fn_span: _ } => todo!(),
+            TailCall { ref func, ref args, fn_span: _ } => {
+                // FIXME(explicit_tail_calls): a lot of code here is duplicated with normal calls, can we refactor this?
```

Comment:
would be nice, do you need anything for trying this out?

Reply:
I think it just works, here is my attempt: 119cb0d. I'm not sure about the naming and it's a bit annoying with all the things we need to pass though

Reply:
That looks good, though I hope we can find a better name for that function. `eval_callee_and_args` or so?

Also the doc comment should say what that `bool` means.

Reply:
`eval_callee_and_args` sounds better. Would it make sense to return a structure (say `EvaluatedCalleeAndArgs`) so that it's harder to misinterpret the return values (because of the field names)?

Reply:
Yeah a structure is a good idea.

Reply:
Renamed & added a struct: 406bd6b